onnx/shape_inference/implementation.cc

```diff
@@ -688,8 +688,28 @@ std::vector<const TypeProto*> GraphInferencerImpl::doInferencing(
   SymbolTable* symbolTable = getSymbolTable();
   int numInputs = int(inputTypes.size());
 
-  if (g_->input_size() != numInputs) {
-    fail_shape_inference("Graph has ", g_->input_size(), " inputs but ", numInputs, " were provided");
+  if (getIRVersion() >= 4) {
+    if (g_->input_size() != numInputs) {
+      fail_shape_inference("Graph has ", g_->input_size(), " inputs but ", numInputs, " were provided");
```

Comment:
Does this properly enforce that none of these inputs have initializers?

Reply:
No it does not and I have added a check for it. Thanks